### PR TITLE
Refactor softmax_cudnn kernel impl for code reuse.

### PR DESCRIPTION
**Pull request by limin2021**
### PR types
Performance optimization
### PR changes
OPs
### Describe
Refactor softmax_cudnn kernel impl for code reuse.
(1) move the cuda kernel impl in softmax_cudnn_op.cu to softmax_cudnn_op.cu.h for code reuse in future fused attention op.

Unittest results of softmax op:
d21f337727b9d62120ebdbf29adfe21c

### CODE REVIEW

**Comment by xingfeng01** [comment]
LGTM. CI 恢复后 rerun 一下.

File: paddle/fluid/operators/softmax_cudnn_op.cu [hunk 1]
```diff
@@ -549,91 +40,11 @@ class SoftmaxGradCUDNNKernel : public framework::OpKernel<T> {
...
-    auto dims = out->dims();
-    const int rank = dims.size();
-    const int axis = CanonicalAxis(ctx.Attr<int>("axis"), rank);
-    const int dim = dims[axis];
-    const int N = SizeToAxis(axis, dims);
-    const int D = SizeOutAxis(axis, dims);
-
-    constexpr int max_dim = 320;
-    constexpr int warps_per_block = 4;
+    // auto* dx_data = dx->data<T>();
```
Comment (zhangting2020) [inline paddle/fluid/operators/softmax_cudnn_op.cu]: 这行注释删掉

Comment (limin2021) [reply paddle/fluid/operators/softmax_cudnn_op.cu]: Done.

**Review by zhangting2020 — DISMISSED** [review]
LGTM

**Review by Xreki — APPROVED** [review]
LGTM

**Pull request merged**